runtime/src/main/java/org/corfudb/runtime/collections/Table.java

```diff
@@ -58,18 +58,35 @@
     @Getter
     private final TableMetrics metrics;
 
+    @Getter
+    private final Class<K> keyClass;
+
+    @Getter
+    private final Class<V> valueClass;
+
+    @Getter
+    private final Class<M> metadataClass;
+
     /**
      * Returns a Table instance backed by a CorfuTable.
      *
-     * @param namespace               Namespace of the table.
-     * @param fullyQualifiedTableName Fully qualified table name.
-     * @param valueSchema             Value schema to identify secondary keys.
-     * @param corfuRuntime            Connected instance of the Corfu Runtime.
-     * @param serializer              Protobuf Serializer.
+     * @param namespace               namespace of the table
+     * @param fullyQualifiedTableName Fully qualified table name
+     * @param kClass                  key class
+     * @param vClass                  value class
```

Comment:
can we derive value and metadata classes from the value and metadata schemas?  Also is the metadataSchema param missing in the comment?

Reply:
Yes we can, that's what I initially did, but later I think it's cleaner to pass them all then doing a  bunch of castings, as these params are the same as what were passed in from the caller.